gr-fft/lib/window.cc

```diff
@@ -50,33 +50,51 @@ double freq(int ntaps) { return 2.0 * GR_M_PI / ntaps; }
 
 double rate(int ntaps) { return 1.0 / (ntaps >> 1); }
 
-double window::max_attenuation(win_type type, double beta)
+double window::max_attenuation(win_type type, double param)
 {
     switch (type) {
     case (WIN_HAMMING):
         return 53;
-        break;
     case (WIN_HANN):
         return 44;
-        break;
     case (WIN_BLACKMAN):
         return 74;
-        break;
     case (WIN_RECTANGULAR):
         return 21;
-        break;
     case (WIN_KAISER):
-        return (beta / 0.1102 + 8.7);
-        break;
+        // linear approximation form
+        return (param / 0.1102 + 8.7);
     case (WIN_BLACKMAN_hARRIS):
         return 92;
-        break;
     case (WIN_BARTLETT):
         return 27;
-        break;
     case (WIN_FLATTOP):
         return 93;
-        break;
+    case WIN_NUTTALL:
+        return 114;
+    case WIN_NUTTALL_CFD:
+        return 112;
+    case WIN_WELCH:
+        return 31;
+    case WIN_PARZEN:
+        return 56;
+    case WIN_EXPONENTIAL:
+        // varies slightly depending on the decay factor, but this is a safe return value
+        return 26;
+    case WIN_RIEMANN:
+        return 39;
+    case WIN_GAUSSIAN:
+        // value not meaningful for gaussian windows, but return something reasonable
+        return 100;
+    case WIN_TUKEY:
+        // low end is a rectangular window, attenuation exponentialy approaches Hann
+        // piecewise linear estimate, will be slightly greater than expected attenuation
+        if (param < 0.7)
+            return (param * 5 + 21);
```

Comment:
Do these magic values have a semantic meaning? If so, it would be nice to have them be named constants.

Reply:
All of that is a empirically derived piecewise linear approximation (orange) due to how the attenuation changes with alpha (blue). The value is not critical, but underestimating it can result in firdes returning a filter with too few coefficients for the desired attenuation. I'll clarify that.

![image](https://user-images.githubusercontent.com/7121282/105608005-9b5ffc00-5d5e-11eb-9821-40d42836c002.png)

Reply:
Yup, worth a comment at least. 👍